src/core/validations/message_test.rs

```diff
@@ -676,4 +676,20 @@ mod tests {
         );
         assert_validation_error(&msg, ValidationError::ExceededMaxStorageUnits);
     }
+
+    #[test]
+    fn test_storage_lend_timestamp_too_far_in_past() {
```

Comment:
Let's add an explicit block engine test for a user replaying an old storage lend message after lender has zeroed out the lent storage.